wagtail/admin/views/pages/listing.py

```diff
@@ -71,10 +86,26 @@ class PageFilterSet(WagtailFilterSet):
         queryset=Site.objects.all(),
         widget=CheckboxSelectMultiple,
     )
+    has_child_pages = HasChildPagesFilter(
+        label=_("Has child pages"),
+        widget=Select(
+            choices=[
+                ("", _("Any")),
+                (True, _("Yes")),
+                (False, _("No")),
+            ]
+        ),
+    )
 
     class Meta:
         model = Page
-        fields = ["content_type", "latest_revision_created_at", "owner", "site"]
+        fields = [
+            "content_type",
+            "latest_revision_created_at",
+            "owner",
+            "site",
+            "has_child_pages",
```

Comment:
We probably shouldn't add this here because it's not an actual field.

django-filter will normally complain with `TypeError: 'Meta.fields' must not contain non-model field names: has_child_pages`, but in this case it's obscured because the filter is overridden by the `has_child_pages = HasChildPagesFilter(...)` above.

While it's unlikely, django-filter might be more aggressive in the future and enforce this even if we have defined a filter override. This is why I suggested using `fields = []` unless we want to auto-generate some filters from the fields :D